cluster/engine.go

```diff
@@ -105,6 +107,7 @@ type Engine struct {
 	networks        map[string]*Network
 	volumes         map[string]*Volume
 	client          dockerclient.Client
+	apiClient       engineapi.APIClient
```

Comment:
I feel like it may be better to wrap {client, apiClient, failureCount, updateAt} into a struct. The first change is to switch dockerclient.Client to this new client. Next changes are mostly inside this new client struct. It may be better to isolate the changes. Additional benefit is to get failureCount integrated.